### PR TITLE
feat: consolidate fee tests

### LINKED ISSUE #15963: Consolidate fee tests
We want to consolidate all tests that check transaction fees in a single place. This will give us a better overview of what we have and what is missing. We will also see if important variants are missing.

1. Create a new suite for each service in `com.hedera.services.bdd.suites.fees`
2. Go through the code base and copy any tests that check fees to the appropriate suite. Simplify the tests so that they only check the fees, removing all other steps and checks that may be there.
3. Add missing tests

**CryptoService**
--
- [x] CryptoCreate
- [x] CryptoAccountAutoRenew  - (not supported)
- [x] CryptoDeleteAllowance
- [x] CryptoApproveAllowance
- [x] CryptoUpdate
- [x] CryptoTransfer
- [x] CryptoTransfer (tokens)
- [x] CryptoTransfer (custom fees)
- [x] CryptoDelete
- [x] CryptoGetAccountRecords
- [x] CryptoGetAccountBalance
- [x] CryptoGetInfo
- [x] CryptoGetStakers - (not supported) - deprecated
- [x] CryptoAddLiveHash - (not supported) - deprecated
- [x] CryptoDeleteLiveHash - (not supported) - deprecated
- [x] CryptoGetLiveHash - (not supported) - deprecated
 
**ConsensusService**
--
- [x] ConsensusCreateTopic
- [x] ConsensusUpdateTopic
- [x] ConsensusDeleteTopic
- [x] ConsensusSubmitMessage
- [x] ConsensusGetTopicInfo

**TokenService**
--
- [x] TokenAirdrop (no prior association with token)
- [x] TokenClaimAirdrop
- [x] TokenCancelAirdrop
- [x] TokenReject
- [x] TokenCreate
- [x] TokenCreate (custom fees)
- [x] TokenUpdate
- [x] TokenFeeScheduleUpdate
- [x] TokenDelete
- [x] TokenAssociate
- [x] TokenDissociate
- [x] TokenMint (fungible)
- [x] TokenMint (non-fungible)
- [x] TokenMint (bulk mint 10k NFTs)
- [x] TokenBurn
- [x] TokenGrantKyc
- [x] TokenRevokeKyc
- [x] TokenFreeze
- [x] TokenUnfreeze
- [x] TokenPause
- [x] TokenUnpause
- [x] TokenWipe
- [x] TokenGetInfo
- [x] TokenGetNftInfo
- [x] TokenGetNftInfos  - not supported
- [x] TokenGetAccountNftInfos  - not supported
- [x] TokenUpdateNfts (updates metadata of 1 NFT)
- [x] TokenUpdateNfts (update multiple NFTs in a single call)

**ScheduleService**
--
- [x] ScheduleCreate
- [x] ScheduleSign
- [x] ScheduleDelete
- [x] ScheduleGetInfo

**FileService**
--
- [x] FileCreate
- [x] FileUpdate
- [x] FileDelete
- [x] FileAppend
- [x] FileGetContents
- [x] FileGetInfo

**Miscellaneous**
--
- [x] PrngTransaction
- [x] GetVersion
- [x] GetByKey (not supported) - deprecated
- [x] TransactionGetReceipt - free query
- [x] TransactionGetRecord

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
Consolidate fee tests and add missing tests

**Related issue(s)**:
Fixes #15963

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/FileServiceFeesSuite.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                                .key(magicWacl)
+                                .lifetime(THREE_MONTHS_IN_SECONDS)
+                                .contents("Nothing much!"))
+                .when(fileAppend(targetFile)
+                        .signedBy(magicKey)
+                        .blankMemo()
+                        .content(contentBuilder.toString())
+                        .payingWith(civilian)
+                        .via(baseAppend))
+                .then(validateChargedUsdWithin(baseAppend, expectedAppendFeesPriceUsd, 0.01));
```
Comment (iwsimon) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/FileServiceFeesSuite.java:65]: Maybe we can remove those given, when and then in the tests?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java [hunk 2]
```diff
@@ -0,0 +1,908 @@
...
+    private static final double EXPECTED_REVOKEKYC_PRICE_USD = 0.001;
+    private static final double EXPECTED_NFT_MINT_PRICE_USD = 0.02;
+    private static final double EXPECTED_FUNGIBLE_MINT_PRICE_USD = 0.001;
+    private static final double EXPECTED_FUNGIBLE_REJECT_PRICE_USD = 0.001;
+    private static final double EXPECTED_NFT_REJECT_PRICE_USD = 0.001;
+    private static final String OWNER = "owner";
+
+    @HapiTest
+    @DisplayName("charge association fee for FT correctly")
+    final Stream<DynamicTest> chargeAssociationFeeForFT() {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:136]: We can include `airdrop` to the title

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java [hunk 3]
```diff
@@ -0,0 +1,908 @@
...
+                tokenAirdrop(moving(1, FUNGIBLE_TOKEN).between(OWNER, receiver))
+                        .payingWith(OWNER)
+                        .via("second airdrop"),
+                validateChargedUsd("airdrop", 0.1, 1),
+                validateChargedUsd("second airdrop", 0.05, 1));
+    }
+
+    @HapiTest
+    @DisplayName("charge association fee for NFT correctly")
+    final Stream<DynamicTest> chargeAssociationFeeForNFT() {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:159]: We can include `airdrop` to the name

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java [hunk 4]
```diff
@@ -0,0 +1,908 @@
...
+                // assert balances NFT
+                getAccountBalance(RECEIVER).hasTokenBalance(NON_FUNGIBLE_TOKEN, 1),
+                // assert token associations
+                getAccountInfo(RECEIVER).hasToken(relationshipWith(FUNGIBLE_TOKEN)),
+                getAccountInfo(RECEIVER).hasToken(relationshipWith(NON_FUNGIBLE_TOKEN))));
+    }
+
+    @HapiTest
+    @DisplayName("FT happy path")
+    final Stream<DynamicTest> ftHappyPath() {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:235]: Same

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:235]: There are a few more below. I won't spam on all of them

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java [hunk 5]
```diff
@@ -0,0 +1,908 @@
...
+import com.hederahashgraph.api.proto.java.TokenSupplyType;
+import com.hederahashgraph.api.proto.java.TokenType;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+
+public class TokenServiceFeesSuite {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:92]: I don't see this test: `TokenMint (bulk mint 10k NFTs)`

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/fees/TokenServiceFeesSuite.java:95]: Per discussion with Lev - the max transfer is 10. The described test is not possible.

**Review by vtronkov — APPROVED** [review]

**Review by vtronkov — APPROVED** [review]

**Review by lukelee-sl — APPROVED** [review]
LGTM. tyvm!

**Pull request merged**